### PR TITLE
ec_deployment: Add remote_cluster to Elasticsearch

### LINKED ISSUE #137: ec_deployment: Support Cross Cluster Search settings
## Overview

Currently, there's no way users can configure the cross cluster search settings in their deployments, it would be great to allow them to do so.

## Possible Implementation

Adding a new `remote_cluster` block which allows users to set one or more remote clusters.

```hcl
resource "ec_deployment" "source_deployment" {
  name = "my_source_ccs"

  region                 = "us-east-1"
  version                = "7.9.2"
  deployment_template_id = "aws-io-optimized-v2"

  elasticsearch {}

  kibana {}

  apm {}
}

resource "ec_deployment" "ccs" {
  name = "ccs deployment"

  region                 = "us-east-1"
  version                = "7.9.2"
  deployment_template_id = "aws-cross-cluster-search-v2"

  elasticsearch {
    remote_cluster {
      deployment_id = ec_deployment.source_deployment.id

      # Remote deployment's elasticsearch ref_id (Optional) defaults to "main-elasticsearch"
      ref_id = "main-elasticsearch"

      # If true, skip this cluster during search if it is disconnected
      skip_unavailable = false

      # The alias for this remote cluster. Aliases must only contain letters, digits, dashes and underscores
      alias = "some alias"
    }
  }

  kibana {}
}
```

## Testing

An acceptance test or two wouldn't be bad.

## Context

Similar to what the UI allows:

image

### PR DESCRIPTION
**Pull request by marclop**
## Description

Adds the ability to configure cross cluster search on the Elasticsearch
resource, allowing users to fully benefit from the cross-cluster-search
deployment template, useless before this patch.

The `remote_cluster` block can be set multiple times, and only requiring
the remote `deployment_id` to be set (defaulting `ref_id` to the default
elasticsearch `ref_id`).

```hcl
resource "ec_deployment" "ccs" {
  name = "ccs deployment"

  region                 = "us-east-1"
  version                = "7.9.2"
  deployment_template_id = "aws-cross-cluster-search-v2"

  elasticsearch {
    remote_cluster {
      deployment_id = ec_deployment.source_deployment.id
      alias         = ec_deployment.source_deployment.name
    }

    remote_cluster {
      deployment_id = ec_deployment.second_source.id
      alias         = ec_deployment.second_source.name
    }
  }

  kibana {}
}
```

The `elastic/cloud-sdk-go` dependency has been updated to the latest
version to be able to use the CCS APIs and merge that horrendous bugfix.

## Related Issues

Closes #137 

## How Has This Been Tested?

Manually, many hours, unit and acceptance tested.

## Types of Changes

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: ec/ecresource/deploymentresource/elasticsearch_remote_cluster_expanders.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+	return esremoteclustersapi.Update(esremoteclustersapi.UpdateParams{
+		API:             client,
+		DeploymentID:    d.Id(),
+		RefID:           d.Get("elasticsearch.0.ref_id").(string),
+		RemoteResources: res,
+	})
+}
+
+func expandRemoteClusters(raw []interface{}) (*models.RemoteResources, error) {
```
Comment (karencfv) [inline ec/ecresource/deploymentresource/elasticsearch_remote_cluster_expanders.go:48]: We don't seem to be returning errors anywhere 🤔 It seems safe to only have `*models.RemoteResources` as a return value on the signature?

File: ec/ecresource/deploymentresource/read.go [hunk 2]
```diff
@@ -46,7 +48,15 @@ func readResource(_ context.Context, d *schema.ResourceData, meta interface{}) d
 		return diag.FromErr(multierror.NewPrefixed("failed reading deployment", err))
 	}
 
-	if err := modelToState(d, res); err != nil {
+	remotes, _ := esremoteclustersapi.Get(esremoteclustersapi.GetParams{
```
Comment (karencfv) [inline ec/ecresource/deploymentresource/read.go:51]: It'll be good to return this error

File: examples/deployment_ccs/README.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Deployment Example
+
+This example shows how to deploy multiple Elastic Cloud deployments and reference one another through Cross Cluster Search using Terraform only.
```
Comment (karencfv) [inline examples/deployment_ccs/README.md:3]:
```suggestion
This example shows how to deploy multiple Elastic Cloud deployments, and reference one another through Cross Cluster Search using Terraform only.
```

**Review by karencfv — COMMENTED** [review]
Looks good! Just left a few minor comments

**Review by karencfv — APPROVED** [review]
Thanks for addressing my comments :)

**Comment by marclop** [comment]
Failing tests are no bother, I'm merging.

**Pull request merged**